### PR TITLE
Cppcheck various

### PR DESCRIPTION
**Pull request by rbiasini**
WIP

### CODE REVIEW

File: board/drivers/llcan.h [hunk 1]
```diff
@@ -80,6 +79,5 @@ void llcan_init(CAN_TypeDef *CAN_obj) {
 void llcan_clear_send(CAN_TypeDef *CAN_obj) {
   CAN_obj->TSR |= CAN_TSR_ABRQ0;
   CAN_obj->MSR &= ~(CAN_MSR_ERRI);
-  CAN_obj->MSR = CAN_obj->MSR;
```
Comment (geohot) [inline board/drivers/llcan.h]: Don't remove this! It clears the register

File: board/drivers/llcan.h [hunk 2]
```diff
@@ -12,7 +12,6 @@ void puts(const char *a);
 
 bool llcan_set_speed(CAN_TypeDef *CAN_obj, uint32_t speed, bool loopback, bool silent) {
   // initialization mode
-  CAN_obj->MCR = CAN_MCR_TTCM | CAN_MCR_INRQ;
```
Comment (geohot) [inline board/drivers/llcan.h]: Why was this removed?

Comment (rbiasini) [reply board/drivers/llcan.h]: cppcheck complains about: `Variable 'CAN_obj->MCR' is reassigned a value before the old one has been used.`

File: board/drivers/llcan.h (hunk 1 above)
Comment (rbiasini) [reply board/drivers/llcan.h]: cppcheck complains about: 
`Redundant assignment of 'CAN_obj->MSR' to itself.`

File: board/drivers/llcan.h (hunk 2 above)
Comment (rbiasini) [reply board/drivers/llcan.h]: See: https://github.com/commaai/panda/pull/243/files#diff-b3438395a72aa43adef2658b43dc64a8L32

**Comment by rbiasini** [comment]
just ignored those checks: https://github.com/commaai/panda/commit/f45dd04c6797e71a9c1db79d8d414a44d8b028c1
